x-pack/plugins/ml/public/components/job_selector/job_selector.js

```diff
@@ -137,6 +140,27 @@ export function JobSelector({
     setNewSelection(selectedIds);
   }, [isFlyoutVisible]); // eslint-disable-line
 
+  useEffect(() => {
+    const handleResize = () => {
+      if (jobs.length > 0 && flyoutEl && flyoutEl.current && flyoutEl.current.flyout) {
+        const tzConfig = config.get('dateFormat:tz');
+        const dateFormatTz = (tzConfig !== 'Browser') ? tzConfig : moment.tz.guess();
+        const derivedWidth = Math.round(flyoutEl.current.flyout.offsetWidth / 4);
```

Comment:
this calculation is only happening on page resize. On initial flyout render the default value of 299px is still being used and so if the page is too narrow the gantt charts will fall off the page.
Also, using 1/4 of the flyout width is what is causing the empty space at right side of the charts. It isn't taking into account the internal padding of the flyout (24px) or the padding of the table.

From my tests, adding an extra 48px (left and right flyout padding) to `derivedWidth` gives a better use of space.
![image](https://user-images.githubusercontent.com/22172091/57290769-06166300-70b6-11e9-95ab-cc372e4c46b7.png)

Reply:
Just saw you also added a comment if this could be percentages sorry for the duplicate :)

Reply:
Ah good one - taking into account the flyout/table padding makes sense when setting the width. I'll make sure the calculation occurs on first render and adjust it.

Re: percentages - yes, that's a good question. I tried using percentages but it doesn't work because the divs are empty (dashed line and the bit showing the time covered). I tried adding a non-breaking space (&nbsp;), setting min-width, etc to get relative widths to work but no such luck. Only specific widths did the trick. 

I'm happy to try any ideas I may have missed.